test/bivariate_statistics_test.cpp

```diff
@@ -38,7 +38,314 @@ using boost::multiprecision::cpp_complex_50;
 using  boost::math::statistics::means_and_covariance;
 using  boost::math::statistics::covariance;
 
-template<class Real>
+#if (__cplusplus > 201700 || _MSVC_LANG > 201700) && (__GNUC__ > 9 || (__clang_major__ > 9 && defined __GLIBCXX__)  || _MSC_VER > 1927)
+#include <execution>
```

Comment:
Shouldn't this only need to be in the library code, not the user code?

Reply:
This is here so that I could keep all the tests in one file. It is one giant if-else macro with a main for C++11&14 and a main for C++17&20. Normal users would not need this.